src/dex/balancer-v2/VirtualBoostedPool.ts

```diff
@@ -0,0 +1,650 @@
+   * @returns Array of VirtualBoosted subgraph pools. Dictionary object of virtualBoosted pool data.
+   */
+  static createPools(pools: SubgraphPoolBase[]): VirtualBoostedPools {
+    const subgraph: SubgraphPoolBase[] = [];
+    const boostedPools: BoostedPools = {};
+    const phantomPools: SubgraphPoolBase[] = [];
+    const linearDict: { [address: string]: SubgraphPoolBase } = {};
+    // Create an initial list of phantom pools and dictionary of Linear pools.
+    pools.forEach(p => {
+      // Covers ERC4626 & Aave Linear Pools
+      if (p.poolType.includes('Linear')) linearDict[p.address] = p;
+      else if (p.poolType === BalancerPoolTypes.StablePhantom) {
+        phantomPools.push(p);
+      }
+    });
+
+    // Create a boostedPool for each phantom that consists of Linear BPTs.
+    phantomPools.forEach(phantomPool => {
+      const mainTokens: MainToken[] = [];
+      const isBoosted = phantomPool.tokens.every(t => {
+        if (t.address === phantomPool.address) return true;
+        // Phantom pools contain their own BPT in tokens list
+        else if (linearDict[t.address]) {
+          // mainToken are the underlying mainTokens of the Linear Pools and will be seen as the tokens the VirtualPool contains.
+          // i.e.bb - a - USD VirtualBoostedPool will look like it has USDC, DAI, USDT and bb - a - USD_bpt
+          mainTokens.push({
+            address:
+              linearDict[t.address].tokens[linearDict[t.address].mainIndex]
+                .address,
+            decimals:
+              linearDict[t.address].tokens[linearDict[t.address].mainIndex]
+                .decimals,
+            linearPoolAddr: t.address,
+            linearPoolId: linearDict[t.address].id,
+          });
+          return true;
+        } else return false; // BoostedPools must consist of Phantom + Linear
+      });
+
+      if (isBoosted) {
+        // phantomStablePoolAddr is the PhantomStablePool connecting all LinearPools
+        boostedPools[phantomPool.id] = {
+          mainTokens,
+          phantomStablePoolAddr: phantomPool.address,
+        };
+        subgraph.push({
+          id: phantomPool.id + this.poolType.toLowerCase(),
+          address: phantomPool.address + this.poolType.toLowerCase(),
+          poolType: this.poolType,
+          tokens: mainTokens,
+          mainIndex: 0,
+          wrappedIndex: 0,
+          totalLiquidity: phantomPool.totalLiquidity,
+        });
+      }
+    });
+
+    return { dictionary: boostedPools, subgraph };
+  }
+
+  static poolType = 'VirtualBoosted';
+  vaultAddress: string;
+  vaultInterface: Interface;
+  linearPoolInterface: Interface;
+  phantomStablePoolInterface: Interface;
+
+  constructor(vaultAddress: string, vaultInterface: Interface) {
+    this.vaultAddress = vaultAddress;
+    this.vaultInterface = vaultInterface;
+    this.linearPoolInterface = new Interface(LinearPoolABI);
+    this.phantomStablePoolInterface = new Interface(MetaStablePoolABI);
```

Comment:
It would be better to make these static members to avoid creating interface for each pool while it can be reused. This can also be fixed in other pool implementation

Reply:
Changed to create in main `BalancerV2EventPool` and reuse when instantiating pools.